v1/README_Cassandra_To_Cloud_Bigtable.md

```diff
@@ -36,6 +36,8 @@ on [Metadata Annotations](https://github.com/GoogleCloudPlatform/DataflowTemplat
 * **defaultColumnFamily** : The name of the column family of the Bigtable table. The default value is default.
 * **rowKeySeparator** : The separator used to build row-keys. The default value is '#'.
 * **splitLargeRows** : The flag for enabling splitting of large rows into multiple MutateRows requests. Note that when a large row is split between multiple API calls, the updates to the row are not atomic. .
+* **writetimeCassandraColumnSchema** : GCS path to schema to copy Cassandra writetimes to Bigtable. The command to generate this schema is ```cqlsh -e "select json * from system_schema.columns where keyspace_name='$CASSANDRA_KEYSPACE' and table_name='$CASSANDRA_TABLE'`" > column_schema.json```. Set $WRITETIME_CASSANDRA_COLUMN_SCHEMA to a GCS path, e.g. `gs://$BUCKET_NAME/column_schema.json`. Then upload the schema to GCS: `gcloud storage cp column_schema.json $WRITETIME_CASSANDRA_COLUMN_SCHEMA`. Requires Cassandra version 2.2 onwards for JSON support.
```

Comment:
> cqlsh -e "select json * from system_schema.columns where keyspace_name='$CASSANDRA_KEYSPACE' and table_name='$CASSANDRA_TABLE'

What happens if you execute this query from the cassandra client and interpret the results from resultset? I find it hard to behelive that we can query using cqlsh but can't fetch this data via an API. 

I have suffered a lot from maintaining compatiblity of hbase clients. I don't want to add similar pain by parsing strings in cassandra. Lets see if there is any way to do this via API contract

Reply:
Discussed offline - will keep the current approach. 

The result of the cqlsh command will be like [the following](https://docs.datastax.com/en/cql-oss/3.x/cql/cql_using/useQueryJSON.html): 
```
cqlsh:cycling> select json name, checkin_id, timestamp from checkin;
 [json]
------------------------------------------------------------------------------------------------------------------
 {"name": "BRAND", "checkin_id": "50554d6e-29bb-11e5-b345-feff8194dc9f", "timestamp": "2016-08-28 21:45:10.406Z"}
  {"name": "VOSS", "checkin_id": "50554d6e-29bb-11e5-b345-feff819cdc9f", "timestamp": "2016-08-28 21:44:04.113Z"}
(2 rows)
```

Note the prefix `[json] ---...` and suffix `(2 rows)` strings that will interfere with strict parsing. I wasn't able to find a library that does fuzzy JSON matching, so we have our own JSON parser for this.

Alternatively, datastax does have a systems querier to get system column table info. The ideal scenario would be to slot a datastax connector as the first step in the pipeline to process Cassandra table schema, and then feed the results to CassandraIO. However:
1. CassandraIO [does not allow any processors before itself in a pipeline](https://beam.apache.org/releases/javadoc/2.29.0/org/apache/beam/sdk/io/cassandra/CassandraIO.Read.html)
2. It is not possible to preprocess steps and dynamically construct a pipeline in a V1 template

In a future iteration, we should either loosen CassandraIO.Read or upgrade this template to a V2 template so we can have the writetimeCassandraColumnSchema automatically processed. Added a TODO for myself to explore these approached at a future time.